dbfit-java/core/src/main/java/dbfit/util/sql/PreparedStatements.java

```diff
@@ -8,11 +8,16 @@ private static String buildParamList(int numberOfParameters) {
         return join(repeat("?", numberOfParameters), ", ");
     }
 
-    public static String buildStoredProcedureCall(String procName, int numberOfPararameters) {
-        return "{ call " + procName + "(" + buildParamList(numberOfPararameters) + ")}";
+    private static String buildReturnValue(boolean returnsValue) {
+        return returnsValue ? "? = " : "";
     }
 
-    public static String buildFunctionCall(String procName, int numberOfParameters) {
-        return "{ ? = call " + procName + "(" + buildParamList(numberOfParameters - 1) + ")}";
+    public static String buildStoredRoutineCallStatementText(
+            String routineName, int numberOfPararameters, boolean hasReturnValueParameter) {
+        String callText = "{ " + buildReturnValue(hasReturnValueParameter);
+        callText += "call " + routineName + "(";
+        callText += buildParamList((hasReturnValueParameter ? numberOfPararameters - 1 : numberOfPararameters));
+        callText += ") }";
+        return callText;
```

Comment:
What if we skip the `callText` temporary variable here?

And yet again, it's a bit hard to grasp the overall call pattern. As soon as it makes things clearer some temp variables would be OK.